swap_meet/vendor.py

```diff
@@ -1,2 +1,57 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self,inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+
+        return [item for item in self.inventory if category == item.category]
+
+    def swap_items(self,friend, my_item, their_item):
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            friend.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            friend.inventory.append(my_item)
+            return True
+        return False
+    
+    def swap_first_item(self, friend):
+
+        if self.inventory and friend.inventory:
+            return self.swap_items(friend, self.inventory[0], friend.inventory[0])
```

Comment:
What would this function return if self.inventory and friend.inventory were empty? 

How could you use a guard clause here to handle that scenario?

Also, it's great to reuse the swap_items method you already wrote! However, swap_items has linear time complexity because it invokes the remove() method.

Can you think of how you could write a custom swapping implementation here to bring the time complexity to O(1) ?